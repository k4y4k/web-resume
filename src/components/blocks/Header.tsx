import * as React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import Subtitle from '../header/Subtitle'
import Title from '../header/Title'
import tw from 'twin.macro'

interface PureHeaderTypes {
  title: string
  subtitle: string
  lightmode: boolean
  compact: boolean
}

interface HeaderTypes {
  lightmode?: boolean
  compact?: boolean
}

export const PureHeader = ({
  title,
  subtitle,
  lightmode = false,
  compact = false,
}: PureHeaderTypes): JSX.Element => (
  <div data-testid='header' tw='text-center'>
    <Title compact={compact} lightmode={lightmode} title={title} />
    <Subtitle compact={compact} lightmode={lightmode} subtitle={subtitle} />
    <hr
      css={[
        tw`block my-4`,
        compact && tw`my-8`,
        lightmode ? tw`bg-white` : tw`bg-black`,
      ]}
    />
  </div>
)

export const Header = ({
  lightmode = false,
  compact = false,
}: HeaderTypes): JSX.Element => {
  const data = useStaticQuery(graphql`
    {
      text: file(extension: { eq: "json" }, name: { eq: "data" }) {
        childDataJson {
          basics {
            name
            label
          }
        }
      }
    }
  `)

  const { name, label } = data.text.childDataJson.basics

  return (
    <PureHeader
      title={name}
      lightmode={lightmode}
      subtitle={label}
      compact={compact}
    />
  )
}

export default Header
