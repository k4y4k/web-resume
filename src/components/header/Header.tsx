import * as React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Name from './Name'
import Subtitle from './Subtitle'
import 'twin.macro'

interface PureHeaderTypes {
  name: string
  subtitle: string
}

export const PureHeader = ({
  name,
  subtitle,
}: PureHeaderTypes): JSX.Element => (
  <header aria-label='header' tw='uppercase'>
    <Name name={name} />
    <Subtitle subtitle={subtitle} />
  </header>
)

export const Header = (): JSX.Element => {
  const data = useStaticQuery(graphql`
    {
      file(extension: { eq: "json" }, name: { eq: "data" }) {
        childDataJson {
          basics {
            name
            label
          }
        }
      }
    }
  `)

  const { name, label } = data?.file.childDataJson.basics

  return <PureHeader name={name} subtitle={label} />
}

export default Header
