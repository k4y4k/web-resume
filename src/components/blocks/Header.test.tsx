/* eslint-disable @typescript-eslint/no-var-requires */

import * as React from 'react'
import { Header, PureHeader } from './Header'
import { render, screen } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'

jest.mock('short-uuid')

describe('<Header />', () => {
  beforeEach(() => {
    // Freeze the generated className.
    const uuid = require('short-uuid')
    uuid.generate.mockImplementation(() => '73WakrfVbNJBaAmhQtEeDv')
  })

  beforeAll(() =>
    (useStaticQuery as jest.Mock).mockReturnValue({
      text: {
        childDataJson: {
          basics: {
            name: 'John Doe',
            label: 'Programmer',
          },
        },
      },
    })
  )

  test('displays header', () => {
    const data = {
      file: {
        childDataJson: {
          basics: {
            name: 'kayak',
            label: 'CEO of supermarket cookie quality debates',
          },
        },
      },
      image: {
        childImageSharp: {
          fluid: {
            base64: '',
            aspectRatio: 1,
            src: '',
            srcSet: '',
            srcWebp: '',
            srcSetWebp: '',
            sizes: '',
          },
        },
      },
    }

    render(
      <PureHeader
        title={data?.file.childDataJson.basics.name}
        subtitle={data?.file.childDataJson.basics.label}
      />
    )

    const name = screen.getByText('kayak')
    expect(name).not.toBeFalsy()

    const subtitle = screen.getByText(
      'CEO of supermarket cookie quality debates'
    )
    expect(subtitle).not.toBeFalsy()

    expect(screen.getByTestId('header')).toMatchSnapshot()
  })

  test('renders OK', () => {
    render(<Header />)

    expect(screen.getByTestId('header')).toMatchSnapshot()
  })
})
