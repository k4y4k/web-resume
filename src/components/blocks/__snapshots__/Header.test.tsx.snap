// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> displays header 1`] = `
<div
  data-testid="header"
>
  <h1
    data-testid="headerTitle"
  >
    kayak
  </h1>
  <h2
    data-testid="headerSubtitle"
  >
    CEO of supermarket cookie quality debates
  </h2>
</div>
`;

exports[`<Header /> renders OK 1`] = `
<div
  data-testid="header"
>
  <h1
    data-testid="headerTitle"
  >
    John Doe
  </h1>
  <h2
    data-testid="headerSubtitle"
  >
    Programmer
  </h2>
</div>
`;
